markdown/3.x/en/documentation/custom-components.md

```diff
@@ -0,0 +1,92 @@
+## Implementing custom components 
+
+---
+FACT-Finder Web Components API can be used to implement custom components in any web framework.
+In a typical scenario, we can use [EventAggregator's](/api/3.x/core-event-aggregator) `addFFEvent()` to query FACT-Finder and [ResultDispatcher's](/api/3.x/core-result-dispatcher) `subscribe()` to receive its response. 
+Below you will find two examples following this approach implementing a product carousel component, one in VueJS and one as a native custom element.
+
+### VueJS
+
+The source code is available [here](https://github.com/FACT-Finder-Web-Components/demos/blob/release/3.x/custom-components/vue-carousel.js).
+Let's take a closer look at the parts where FACT-Finder Web Components API is utilized:
+
+#### 1. `addFFEvent`
+
+We call `addFFEvent()` in the `mounted` hook to trigger FACT-Finder search when the component renders for the first time:
+```javascript
+eventAggregator.addFFEvent({
+    type: 'search',
+    query: carousel.query,
+    topics: function () {
+        return ['carousel-' + carousel._uid];
+    }
+});
```

Comment:
Also adding the `mounted()` wrapper could make it easier to get the context. Especially for folks who just parse for code snippets and only then, if at all, read the surrounding information.
```suggestion
mounted() {
    eventAggregator.addFFEvent({
        type: 'search',
        query: carousel.query,
        topics: function () {
            return ['carousel-' + carousel._uid];
        }
    });
}
```

Reply:
Hmm, unfortunately, it's not that simple in reality. I wanted these snippets to stay in sync with the actual working demo I linked above. So we also have to ensure that ff-core is ready and `ff-communication` settings were already processed. The method we use there is very likely going to be changed in future.

Reply:
I see what you mean. This might indeed cause a huge maintenance overhead that wouldn't be justified by the reward.
Let's keep it as it is.